### PR TITLE
Refactored backgroundStyle to split out backgroundAndTextColors

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Refactored backgroundStyle to split out backgroundAndTextColors.
This will make it easier to share the colors between properties and styles.
The snapshot changes are primarily due to the new code only setting `background-color` instead of `background` when we are only providing a color value.

#### What testing has been done on this PR?

Storybook

#### How should this be manually tested?

Storybook

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

no

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/Grommet/__tests__/Grommet-test.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ describe('Grommet', () => {
 
   test('background', () => {
     const component = renderer.create(
-      <Grommet full background="blue">
+      <Grommet full background="#0000ff">
```
Comment (ShimiSun) [inline src/js/components/Grommet/__tests__/Grommet-test.js:88]: why is this needed? for easier comparison with the snapshot result?

**Comment by ericsoderberghp** [comment]
> What was the drive for these changes?

It will help the Button changes so the Button can set the icon color to match the label color, but with those colors driven more by the theme.

File: src/js/components/Video/__tests__/__snapshots__/Video-test.js.snap [hunk 2]
```diff
@@ -158,7 +158,7 @@ exports[`Video autoPlay renders 1`] = `
 
 .c5:hover {
   background-color: rgba(221,221,221,0.4);
-  color: #000000;
+  color: #FFFFFF;
```
Comment (ShimiSun) [inline src/js/components/Video/__tests__/__snapshots__/Video-test.js.snap:161]: I'm guessing this is the desired behavior (switching the color from black to white), but wondering what would be the impact on the users.

File: src/js/components/Grommet/__tests__/Grommet-test.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/Grommet/__tests__/Grommet-test.js:88]: We are now more accurate setting the text color based on the background color. For example, if you give a background color of "red", we don't know whether to choose light or dark text color, so we leave the text color unset in that case.

I've updated the stories and tests to use hex color values more often, so we can check the mapping more. But, there are still some cases where we leave CSS color names, so we can verify that we don't set a corresponding text color in those cases.

File: src/js/components/Video/__tests__/__snapshots__/Video-test.js.snap (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/Video/__tests__/__snapshots__/Video-test.js.snap:161]: I verified interacting with Video that it's behaving correctly. And I couldn't find any other situations that seemed like a change wasn't correct.
